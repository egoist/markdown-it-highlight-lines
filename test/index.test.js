import Markdown from 'markdown-it'
import highlightLines from '../src'

test('main', () => {
  const md = new Markdown()
  md.use(highlightLines)
  const html = md.render(`
\`\`\`js
const a = b
\`\`\`

\`\`\`js{1}
const a = b
\`\`\`

\`\`\` js {2-3}
const a = b
const c = d
const d = e
\`\`\`
  `)

  expect(html).toMatchSnapshot()
})

test('respect custom <pre>', () => {
  const md = new Markdown({
    highlight(code, language) {
      return `<pre class="language-${language}"><code class="language-${language}">${code}</code></pre>`
    }
  })
  md.use(highlightLines)
  const html = md.render(`
\`\`\`js
const a = b
\`\`\`

\`\`\`js{1}
const a = b
\`\`\`

\`\`\` js {2-3}
const a = b
const c = d
const d = e
\`\`\`
  `)

  expect(html).toMatchSnapshot()
})
