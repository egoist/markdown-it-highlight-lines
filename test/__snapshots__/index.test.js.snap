// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main 1`] = `
"<pre><code class=\\"language-js\\">const a = b
</code></pre>
<pre class=\\"language-js\\"><code class=\\"language-js\\"><span class=\\"highlighted-line\\">const a = b</span></code></pre><pre class=\\"language-js\\"><code class=\\"language-js\\">const a = b
<span class=\\"highlighted-line\\">const c = d</span><span class=\\"highlighted-line\\">const d = e</span></code></pre>"
`;

exports[`respect custom <pre> 1`] = `
"<pre class=\\"language-js\\"><code class=\\"language-js\\">const a = b
</code></pre>
<pre class=\\"language-js\\"><code class=\\"language-js\\"><span class=\\"highlighted-line\\"><pre class=\\"language-js\\"><code class=\\"language-js\\">const a = b</span></code></pre></code></pre><pre class=\\"language-js\\"><code class=\\"language-js\\">const a = b
<span class=\\"highlighted-line\\">const c = d</span><span class=\\"highlighted-line\\">const d = e</span></code></pre>
"
`;
